packages/@lwc/template-compiler/src/codegen/helpers.ts

```diff
@@ -246,13 +250,27 @@ function isStaticNode(node: BaseElement): boolean {
                 isFragmentOnlyUrl(value.value as string)
             )
         );
-    }); // all attrs are static
-    result &&= directives.length === 0; // do not have any directive
-    result &&= properties.every((prop) => isLiteral(prop.value)); // all properties are static
+    });
+
+    // all directives are static-safe
+    result &&= !directives.some((directive) => !STATIC_SAFE_DIRECTIVES.has(directive.name));
+
+    // all properties are static
+    result &&= properties.every((prop) => isLiteral(prop.value));
 
     return result;
 }
 
+function isSafeStaticChild(childNode: ChildNode) {
+    if (!isBaseElement(childNode)) {
+        // don't need to check non-base-element nodes, because they don't have listeners/directives
+        return true;
+    }
+    // Bail out if any children have event listeners or directives. These are only allowed at the top level of a
+    // static fragment, because the engine currently cannot set listeners/refs/etc. on nodes inside a static fragment.
```

Comment:
Do we plan on applying event listeners/directives to the children of static fragments in the future?

Reply:
Probably not, because it'd be pretty complex to reach inside the fragment and figure out where to apply the listeners/directives.